Weno fixes and validations

sjpadgett opened the pull request:
- numerous warnings


<!-- add that issue number that is fixed by this PR (In the form Fixes #123) -->
Fixes #

#### Short description of what this resolves:
- Plan is to add show or not various setup items based on what credentials are set. Also a help when module is enabled.
- I am really inclined to get rid of Globals setup and move to module. Now with new actions listener we are able to auto popup setups or helps.
- I'm planning to modify module manager to auto open config on enable. This way if a setup exist it is presented to user w/o knowing to click the config button. Or I may be able to read a flag in modules config to do it. If not liked, tell me now.

Suddenly my windows doesn't show PHP warnings! It's weird as was working till upgrade to PHP 8.2.

sjpadgett commented:
I'm starting to cancel tests until I'm ready to merge PR. Too me save run times

bradymiller commented:
![2DV](https://github.com/openemr/openemr/assets/278968/ebd335c1-4846-4998-9f8a-6399968056ee)

sjpadgett commented:
Lots going on in last push. Hopefully up for grabs will reset tonight so testing can continue.
Still need to fix config panel style

![image](https://github.com/openemr/openemr/assets/1859985/8a746069-2a7e-4c1c-8317-95099a32c4b0)

sjpadgett commented:
This is some pretty cool stuff if anyone is watching.
- added new manager feature that allows a flag `mod_ui_active` that when set indicates custom styles for buttons if your module is enabled but not fully configure. Though module is enabled it is expected subscriber events be disabled and enable button will act as if module is disabled. See my BootService.
- I added a refresh if action listener return an error after alert is closed.
- Restyled buttons with font awesome.
- Fixed module config container to be responsive and follow themes.

![image](https://github.com/openemr/openemr/assets/1859985/4873eb75-5801-4dc4-aa12-d6a2fa12018f)

sjpadgett commented:
![image](https://github.com/openemr/openemr/assets/1859985/b3e8107f-63c3-4866-9824-d0c438454454)

sjpadgett commented:
New links to open dialogs and setup help

![image](https://github.com/openemr/openemr/assets/1859985/7743657c-76dd-410a-9ab7-3ff6ac062e49)

![image](https://github.com/openemr/openemr/assets/1859985/0ef91fed-4e3d-4ccf-804a-49f17c243d02)

sjpadgett commented:
Also restyle demo choices pharmacy
![image](https://github.com/openemr/openemr/assets/1859985/b1b0da78-28cb-41ac-9766-9bebbc22b212)

bradymiller commented on interface/modules/custom_modules/oe-module-weno/scripts/file_download.php:
```diff
@@ -185,25 +191,42 @@ function download_zipfile($fileUrl, $zipped_file)
+        "pharmacy_log",
+        $_SESSION['authUser'],
+        $_SESSION['authProvider'],
+        1,
```
0 here for the fail

bradymiller commented on interface/modules/custom_modules/oe-module-weno/scripts/weno_log_sync.php:
```diff
@@ -1,38 +1,62 @@
+            "pharmacy_background",
+            $_SESSION['authUser'],
+            $_SESSION['authProvider'],
+            3,
```
just checking why success setting is 3 in 2 calls to this function above

sjpadgett replied:
experimenting.  really need to have event and category different names. not possible that I see. i.e pharmacy_background and pharmacy_download. will look closer at method.

bradymiller commented on interface/modules/custom_modules/oe-module-weno/src/Services/DownloadWenoPharmacies.php:
```diff
@@ -74,28 +68,26 @@ public function extractFile($path_to_extract, $storelocation)
                 return "Failed";
             }
         } else {
             EventAuditLogger::instance()->newEvent("prescriptions_log", $_SESSION['authUser'], $_SESSION['authProvider'], 1, "Failed to extract the file.");
```
should this 1 success code also be 0 here?

bradymiller commented on interface/modules/custom_modules/oe-module-weno/src/Services/LogProperties.php:
```diff
@@ -162,36 +161,43 @@ public function logSync()
             $wenolog->insertWenoLog("prescription", "Success");
         } else {
+            // yes record failures.
             EventAuditLogger::instance()->newEvent("prescriptions_log", $_SESSION['authUser'], $_SESSION['authProvider'], 1, "$statusCode");
```
looks like another spot where should change to 0 success code

bradymiller commented on interface/modules/custom_modules/oe-module-weno/src/Services/TransmitProperties.php:
```diff
@@ -30,23 +34,124 @@ class TransmitProperties
+                    // Append error with icon and onclick event
+                    $uid = attr_js($_SESSION['authUserID'] ?? 0);
+                    $action = attr_js($action);
+                    $error[$type] .= "<i onclick='renderDialog($action, $uid, event)' role='button' class='fas fa-info-circle mx-1'></i>$v<br>";
```
we have an escaping master :+1: 

![kung-fu-baby-funny](https://github.com/openemr/openemr/assets/278968/6c4b00fe-c896-4c29-9c16-20b9935dc9bf)

bradymiller commented on interface/modules/custom_modules/oe-module-weno/templates/weno_fragment.php:
```diff
@@ -6,53 +6,80 @@
+<?php }
+if ($hasWarnings || $hasErrors) { ?>
+    <div id="error-alert" class="alert <?php echo !$justWarnings ? 'alert-danger' : 'alert-warning'; ?> mt-2 px-0 py-1" role="alert">
+        <span class="text-warning"><strong><?php echo text("Problems!"); ?></strong></span> <span style="color: white;"><?php echo xlt("Weno eRx is not fully configured. Details"); ?></span>
```
use xlt instead of text call for the Problems! string

bradymiller commented on interface/modules/custom_modules/oe-module-weno/templates/weno_setup.php:
```diff
@@ -0,0 +1,164 @@
+                        text('3. Find top Patient Bar User icon and click Settings. Scroll down or find the Weno button in left sidebar and click. Enter your email and password in the Weno Provider Email and Weno Provider Password fields and Save.') . '<br /><br />' .
+                        '<h5>' . text('Patient Chart Requirements.') . '</h5>' .
+                        text('1. Each Patient is required to have an assigned primary pharmacy from Demographics->Choices. It is good practice to assign an Alternate Pharmacy too.') . '<br />' .
+                        text('2. Each Patient is required to have a Vitals Height and Weight assigned. Create or enter in from an encounter vitals form') . '<br /><br />';
```
did you mean for above text calls to be xlt instead?

sjpadgett replied:
Not sure I want to xlate. thinking..... ... .. .

sjpadgett commented:
![image](https://github.com/openemr/openemr/assets/1859985/4d6e5994-7c7f-4d6c-b256-5ffa8b08d51c)

sjpadgett commented:
Add status to UI

![image](https://github.com/openemr/openemr/assets/1859985/fa71a97e-a90b-4554-a5fa-f21e5d76a76d)


![image](https://github.com/openemr/openemr/assets/1859985/32977590-fe7b-40a3-ade8-7067c87d96c5)

sjpadgett commented:
@adunsulag @juggernautsei 
After this PR is merged(hopefully by Monday) The ModuleManagerAfterActionListener class and file have been renamed to ModuleManagerListener to better reflect purpose since I added preaction events(preenable, preinstall etc)

So take necessary action for your modules.

juggernautsei commented on interface/modules/custom_modules/oe-module-weno/version.php:
I see you are pitting a version file to assist with patching modules?

sjpadgett replied:
yep there's a current workfow to upgrade but needs work
